tests/resources/aria-at-test-io-format.mjs

```diff
@@ -0,0 +1,1104 @@
+
+    /** @type {SubmitResultDetailsJSON} */
+    const details = {
+      name: state.info.description,
+      task: state.info.task,
+      specific_user_instruction: behavior.specificUserInstruction,
+      summary: {
+        1: {
+          pass: countAssertions(({priority, result}) => priority === 1 && result === CommonResultMap.PASS),
+          fail: countAssertions(({priority, result}) => priority === 1 && result !== CommonResultMap.PASS),
+        },
+        2: {
+          pass: countAssertions(({priority, result}) => priority === 2 && result === CommonResultMap.PASS),
+          fail: countAssertions(({priority, result}) => priority === 2 && result !== CommonResultMap.PASS),
+        },
+        unexpectedCount: countUnexpectedBehaviors(({checked}) => checked),
+      },
+      commands: state.commands.map(command => ({
+        command: command.description,
+        output: command.atOutput.value,
+        support: commandSupport(command),
+        assertions: [...command.assertions, ...command.additionalAssertions].map(assertionToAssertion),
+        unexpected_behaviors: command.unexpected.behaviors
+          .filter(({checked}) => checked)
+          .map(({description, more}) => (more ? more.value : description)),
+      })),
+    };
+
+    /** @type {SubmitResultStatusJSON} */
+    const status = state.commands.map(commandSupport).some(support => support === CommandSupportJSONMap.FAILING)
+      ? StatusJSONMap.FAIL
+      : StatusJSONMap.PASS;
+
+    return {
+      test: state.info.description,
+      details,
+      status,
+    };
+
+    function commandSupport(command) {
+      const allAssertions = [...command.assertions, ...command.additionalAssertions];
+      return allAssertions.some(({priority, result}) => priority === 1 && result !== CommonResultMap.PASS) ||
+        command.unexpected.behaviors.some(({checked}) => checked)
+        ? CommandSupportJSONMap.FAILING
+        : allAssertions.some(({priority, result}) => priority === 2 && result !== CommonResultMap.PASS)
+        ? CommandSupportJSONMap.ALL_REQUIRED
+        : CommandSupportJSONMap.FULL;
+    }
+
+    /**
+     * @param {(assertion: TestRunAssertion | TestRunAdditionalAssertion) => boolean} filter
+     * @returns {number}
+     */
+    function countAssertions(filter) {
+      return state.commands.reduce(
+        (carry, command) => carry + [...command.assertions, ...command.additionalAssertions].filter(filter).length,
+        0
+      );
+    }
+
+    /**
+     * @param {(behavior: TestRunUnexpected) => boolean} filter
+     * @returns {number}
+     */
+    function countUnexpectedBehaviors(filter) {
+      return state.commands.reduce((carry, command) => carry + command.unexpected.behaviors.filter(filter).length, 0);
+    }
+
+    /**
+     * @param {TestRunAssertion | TestRunAdditionalAssertion} assertion
+     * @returns {SubmitResultAssertionsJSON}
+     */
+    function assertionToAssertion(assertion) {
```

Comment:
The name of this function is throwing me off a bit - what is this doing?

Reply:
This function transforms the runtime assertion state to the submitted assertion. The runtime assertion, `TestRunAssertion` and `TestRunAdditionalAssertion`, is the state used and updated by `TestRun`. The submitted assertion, `SubmitResultAssertionsJSON`, becomes a part of `SubmitResultJSON`.